src/utils/rm.js

```diff
@@ -0,0 +1,20 @@
+import { join } from './join'
+
+export async function rm(fs, filepath, opts = {}) {
+  const entries = await fs.readdir(filepath)
+  if (entries == null) {
+    await fs.rm(filepath)
+  } else if (entries.length && opts.recursive) {
+    await Promise.all(
+      entries.map(entry => {
+        const subpath = join(filepath, entry)
+        return fs.lstat(subpath).then(stat => {
+          if (!stat) return
+          return stat.isDirectory() ? rm(fs, subpath, opts) : fs.rm(subpath)
```

Comment:
What about using `fs.unlink` so we are sure that we don't use any logic in FileSystem rm and end up with a recursive call, it may work but it's hard to unwrap this in head. And It will also work with symbolic links.

Reply:
I'm pretty sure that the code already supports removing symlinks. But regardless, that's outside the scope of this issue. The goal of this issue to make recursive rm more robust (for a gitdir that isomorphic-git itself created). I think we should stay focused on that goal.